src/codegen/compvars.cpp

```diff
@@ -1476,7 +1498,8 @@ class NormalObjectType : public ConcreteCompilerType {
             if (rtattr->cls == function_cls) {
                 CompilerVariable* clattr = new ConcreteCompilerVariable(
                     typeFromClass(function_cls), embedRelocatablePtr(rtattr, g.llvm_value_type_ptr), false);
-                return InstanceMethodType::makeIM(var, clattr);
+
+                return InstanceMethodType::makeIM(var, clattr, getClassCompilerVariable(emitter, var));
```

Comment:
We could replace `getClassCompilerVariable()` with `new ConcreteCompilerVariable(UNKNOWN, embedRelocatablePtr(cls, g.llvm_value_type_ptr), false)`